ingenialink/ethercat/network.py

```diff
@@ -454,16 +464,22 @@ def __init_nodes(self, *, release_gil: Optional[bool] = None) -> None:
         nodes = self._ecat_master.config_init(release_gil=release_gil)
         self._lock.release()
         if len(self.servos):
-            self._change_nodes_state(self.servos, pysoem.PREOP_STATE)
+            self._change_nodes_state(
```

Comment:
I think `_change_nodes_state` and `_check_node_state` should return `False` if `servo.slave_exists` is not true. If these functions are called with a node that doesn't have a slave, they will crash. Technically, these functions return `False` if not all the slaves reach the target state, which is correct.